### DISCUSSION TITLE
Perfromance of FINN generated design

### DISCUSSION THREAD

**Discussion by pkdeep** [Xilinx/finn#1063] [Q&A] [answered]
I have a few specific questions regarding the Latency performance of the design generated by FINN:
1. Does pruning impact Latency performance or resource utilization?
2.  Does keeping "Bias" in float format impact performance?
3. How is the AXI Stream width limitation of 8192  calculated?:
    3.1 Like if I have 512x256 in my first layer, the input dimension is 512, and is quantized to 4 bits, and weight and activation are both 2-bits. How will the AXI Stream connection width be calculated:  both in the case of decoupled memory and const mode?

Regards
Pradeep

**Comment by fpjentzsch** [comment, marked as answer]
Hi,
1) Only resource utilization (mostly for high parallelism). If you prune by actually reducing the number of channels in the model, or increase SIMD or PE because the reduced resource utilization gives you more room to parallelize, then it could also impact latency/throughput.
2) No
3) See the following functions to compute the stream widths: https://github.com/Xilinx/finn/blob/39fb8859fec0e47276ffadcafe43092d1b10af7e/src/finn/custom_op/fpgadataflow/matrixvectoractivation.py#L262
